Bottom: 707a09faefc0ab9f4339d4d2d9de51c28d1ba7a3
Top:    cc346790d9fd691fdce657c3a9b4c33e94343c97
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-11 19:35:51 +0000

Remove unused HID method invalidate_wh()

invalidate_lr and invalidate_all are sufficient APIs for our needs.


---

diff --git a/src/hid.h b/src/hid.h
index 3c23338..0e0c479 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -303,12 +303,7 @@ typedef enum
        names behind.  */
     void (*parse_arguments) (int *argc_, char ***argv_);
 
-    /* This may be called outside of redraw to force a redraw.  Pass
-       zero for "last" for all but the last call before control returns
-       to the user (pass nonzero the last time).  If determining the
-       last call is difficult, call *_wh at the end with width and
-       height zero.  */
-    void (*invalidate_wh) (int x_, int y_, int width_, int height_, int last_);
+    /* This may be called to ask the GUI to force a redraw of a given area */
     void (*invalidate_lr) (int left_, int right_, int top_, int bottom_,
 			   int last_);
     void (*invalidate_all) (void);
diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index c10b280..1baa2d8 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -191,11 +191,6 @@ batch_parse_arguments (int *argc, char ***argv)
 }
 
 static void
-batch_invalidate_wh (int x, int y, int width, int height, int last)
-{
-}
-
-static void
 batch_invalidate_lr (int l, int r, int t, int b, int last)
 {
 }
@@ -471,7 +466,6 @@ HID batch_gui = {
   batch_get_export_options,
   batch_do_export,
   batch_parse_arguments,
-  batch_invalidate_wh,
   batch_invalidate_lr,
   batch_invalidate_all,
   batch_set_layer,
diff --git a/src/hid/bom/bom.c b/src/hid/bom/bom.c
index e593bdd..04ad123 100644
--- a/src/hid/bom/bom.c
+++ b/src/hid/bom/bom.c
@@ -544,7 +544,6 @@ HID bom_hid = {
   bom_get_export_options,
   bom_do_export,
   bom_parse_arguments,
-  0,				/* bom_invalidate_wh */
   0,				/* bom_invalidate_lr */
   0,				/* bom_invalidate_all */
   0,				/* bom_set_layer */
diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index bf53c14..a0dbf6b 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -184,8 +184,6 @@ static HID extents_hid = {
   0 /* extents_get_export_options */ ,
   0 /* extents_do_export */ ,
   0 /* extents_parse_arguments */ ,
-
-  0 /* extents_invalidate_wh */ ,
   0 /* extents_invalidate_lr */ ,
   0 /* extents_invalidate_all */ ,
   extents_set_layer,
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 51c95cb..3e5e9db 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -50,12 +50,6 @@ nogui_parse_arguments (int *argc, char ***argv)
 }
 
 static void
-nogui_invalidate_wh (int x, int y, int width, int height, int last)
-{
-  CRASH;
-}
-
-static void
 nogui_invalidate_lr (int l, int r, int t, int b, int last)
 {
   CRASH;
@@ -359,7 +353,6 @@ HID hid_nogui = {
   nogui_get_export_options,
   nogui_do_export,
   nogui_parse_arguments,
-  nogui_invalidate_wh,
   nogui_invalidate_lr,
   nogui_invalidate_all,
   nogui_set_layer,
@@ -417,7 +410,6 @@ apply_default_hid (HID * d, HID * s)
   AD (get_export_options);
   AD (do_export);
   AD (parse_arguments);
-  AD (invalidate_wh);
   AD (invalidate_lr);
   AD (invalidate_all);
   AD (set_layer);
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index afa6a08..dee7d06 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -1079,7 +1079,6 @@ static HID gerber_hid = {
   gerber_get_export_options,
   gerber_do_export,
   gerber_parse_arguments,
-  0 /* gerber_invalidate_wh */ ,
   0 /* gerber_invalidate_lr */ ,
   0 /* gerber_invalidate_all */ ,
   gerber_set_layer,
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 81f0d77..330e716 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -317,13 +317,6 @@ ghid_get_export_options (int *n_ret)
   return NULL;
 }
 
-
-void
-ghid_invalidate_wh (int x, int y, int width, int height, int last)
-{
-  ghid_invalidate_all ();
-}
-
 void
 ghid_invalidate_lr (int left, int right, int top, int bottom, int last)
 {
@@ -1128,8 +1121,6 @@ HID ghid_hid = {
   ghid_get_export_options,
   ghid_do_export,
   ghid_parse_arguments,
-
-  ghid_invalidate_wh,
   ghid_invalidate_lr,
   ghid_invalidate_all,
   ghid_set_layer,
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 84e095a..8d1354f 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -2828,7 +2828,7 @@ lesstif_need_idle_proc ()
 }
 
 static void
-lesstif_invalidate_wh (int x, int y, int width, int height, int last)
+lesstif_invalidate_lr (int l, int r, int t, int b, int last)
 {
   if (!last || !window)
     return;
@@ -2837,16 +2837,10 @@ lesstif_invalidate_wh (int x, int y, int width, int height, int last)
   need_idle_proc ();
 }
 
-static void
-lesstif_invalidate_lr (int l, int r, int t, int b, int last)
-{
-  lesstif_invalidate_wh (l, t, r - l + 1, b - t + 1, last);
-}
-
 void
 lesstif_invalidate_all (void)
 {
-  lesstif_invalidate_wh (0, 0, PCB->MaxWidth, PCB->MaxHeight, 1);
+  lesstif_invalidate_lr (0, 0, PCB->MaxWidth, PCB->MaxHeight, 1);
 }
 
 static int
@@ -3773,7 +3767,6 @@ HID lesstif_gui = {
   lesstif_do_export,
   lesstif_parse_arguments,
 
-  lesstif_invalidate_wh,
   lesstif_invalidate_lr,
   lesstif_invalidate_all,
   lesstif_set_layer,
diff --git a/src/hid/lpr/lpr.c b/src/hid/lpr/lpr.c
index e47c40f..aa696be 100644
--- a/src/hid/lpr/lpr.c
+++ b/src/hid/lpr/lpr.c
@@ -121,7 +121,6 @@ HID lpr_hid = {
   lpr_get_export_options,
   lpr_do_export,
   lpr_parse_arguments,
-  0 /* lpr_invalidate_wh */ ,
   0 /* lpr_invalidate_lr */ ,
   0 /* lpr_invalidate_all */ ,
   0 /* lpr_set_layer */ ,
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index dc1cee5..8e5ff0c 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -1056,7 +1056,6 @@ HID             nelma_hid = {
 	nelma_get_export_options,
 	nelma_do_export,
 	nelma_parse_arguments,
-	0 /* nelma_invalidate_wh */ ,
 	0 /* nelma_invalidate_lr */ ,
 	0 /* nelma_invalidate_all */ ,
 	nelma_set_layer,
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index 2e8e40c..404eb53 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1475,7 +1475,6 @@ HID png_hid = {
   png_get_export_options,
   png_do_export,
   png_parse_arguments,
-  0 /* png_invalidate_wh */ ,
   0 /* png_invalidate_lr */ ,
   0 /* png_invalidate_all */ ,
   png_set_layer,
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index b0e7a7d..0c9345b 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -619,7 +619,6 @@ static HID eps_hid = {
   eps_get_export_options,
   eps_do_export,
   eps_parse_arguments,
-  0 /* eps_invalidate_wh */ ,
   0 /* eps_invalidate_lr */ ,
   0 /* eps_invalidate_all */ ,
   eps_set_layer,
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 3a92dc3..5287518 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1186,7 +1186,6 @@ HID ps_hid = {
   ps_get_export_options,
   ps_do_export,
   ps_parse_arguments,
-  0 /* ps_invalidate_wh */ ,
   0 /* ps_invalidate_lr */ ,
   0 /* ps_invalidate_all */ ,
   ps_set_layer,
